### DISCUSSION TITLE
Solicitud asesoria - Proyecto Entrega 1

### DISCUSSION THREAD

**Discussion by rosapuertac** [rramosp/ai4eng.v1#282] [00 - General]
Hola!
Espero que se encuentre bien.

Hemos seleccionado un dataset para nuestro proyecto, y desarrollamos lo requerido para la primera entrega del proyecto. Sin embargo, tenemos algunas dudas sobre el problema predictivo que elegimos para resolver (El cual formulamos/planteamos nosotros, pues no estaba propuesto en Kaggle). Tememos que en el futuro nos enfrentemos a desafíos que puedan ser demasiado difíciles de superar si no se aclaran ahora. Por lo tanto, nos gustaría solicitar una breve sesión de asesoría de 15 a 30 minutos a través de Google Meet para mañana, 25 de septiembre de 2023, si es posible. 

Agradecemos su consideración y esperamos contar con su orientación para avanzar con confianza en nuestro proyecto.

**Comment by Felipe-RA** [comment]
Saludos @rosapuertac 

Lamentablemente no tengo espacios de asesoría disponibles en este momento, el motivo es porque esta muy cerca de las fechas de entrega y ya otros estudiantes me los tienen reservados.

Ya para el miercoles se te pasa el plazo, así que hagamos lo siguiente:

Ponme por acá toda la información relevante de tu proyecto para poderte ayudar de forma asincrónica.

¡Estoy a la espera, hasta pronto!

**Reply by rosapuertac** [reply]
El objetivo de nuestro proyecto consiste en predecir la evolución futura de la pandemia, a partir del desarrollo de un modelo capaz de proporcionar predicciones precisas sobre el número de casos nuevos y posiblemente la tasa de mortalidad del COVID-19 en diferentes países o regiones. El objetivo central sería construir un modelo que pueda prever con precisión el número de casos nuevos de COVID-19 en el futuro.

El dataset a utilizar se encuentra en un desafío de Kaggle (https://www.kaggle.com/datasets/georgesaavedra/covid19-dataset?resource=download) el cual presenta información de la enfermedad del COVID-19, recopilada en diferentes regiones del mundo. El dataset está compuesto por un conjunto de datos .csv, el cual se caracteriza por presentar un tamaño de 166326 filas y 67 columnas, en las cuales se tiene información de la situación de la pandemia en una región en específico, como es el caso del total de muertes, total de vacunados, pacientes hospitalizados, entre otros. 

En los datos más relevantes se encuentra los siguientes: 
o	location: Contiene el país a donde está asociado el registro 
o	date: Contiene la fecha del registro 
o	total_cases: Contiene el conteo de casos positivos totales hasta el momento 
o	new_cases: Contiene la información de nuevos casos de Covid reportados. 
o	total_deaths: Contiene la información de las muertes causadas por el covid hasta ese día. 
o	new_deaths: Contiene la información de las nuevas muertes reportadas en esa fecha. 
o	icu_patients: Contiene el conteo de pacientes en UCI hasta ese día. 
o	hosp_patients: Contiene el conteo de pacientes que han sido hospitalizados hasta el momento. 
o	people_vaccinated: Contiene el conteo de personas que han sido vacunadas hasta el momento. 
o	new_vaccinations: Contiene el número de nuevos vacunados ese día. 

Tememos que el problema predictivo a resolver sea demasiado ambicioso y que en el futuro nos enfrentemos a desafíos que puedan ser demasiado difíciles de superar.


[owid-covid-data.csv](https://github.com/rramosp/ai4eng.v1/files/12718811/owid-covid-data.csv)

**Reply by Felipe-RA** [reply]
Saludos @rosapuertac 

Veo que dices que tiene 166326 filas. Pero yo veo el dataset sólo con 238. Podrias elaborar de donde obtienes este numero? Tienes que haber claridad de dónde obtienes los datos en tu documento, pues el csv no muestra más que 238x67.

PD. También, no veo que hayas definido la metrica de desempeño, ni un desempeño esperado en producción. Revisa a profundidad el ejemplo de la plataforma

**Reply by rosapuertac** [reply]
Creo que adjunte el dataset equivocado. Disculpa. https://www.kaggle.com/datasets/georgesaavedra/covid19-dataset?resource=download
![image](https://github.com/rramosp/ai4eng.v1/assets/144567194/65d436f8-f41c-4576-b393-21bac5f7c839)

Como metrica de machine learning vamos a usar la matriz de confusion, donde se coteja la información real con la información predicha, en la búsqueda de establecer qué tan fiable o no es el modelo.
Nuestro modelo debería de tener una exactitud igual o superior al 90% y una precisión mayor o igual al 85%, esto debido a que es  una patología grave y es preferible no fallar en la prediccion del número de casos nuevos de COVID-19 en el futuro, pues en una aplicación real esta informacion es crucial para la planificación de recursos de salud, la toma de decisiones gubernamentales y la preparación para posibles aumentos en la propagación del virus. 
Si el modelo no tiene una exactitud del 90% y una precisión del 85%, no vale la pena ponerlo en producción, debido a la poca fiabilidad de este. 

[owid-covid-data.csv](https://github.com/rramosp/ai4eng.v1/files/12720353/owid-covid-data.csv)

**Reply by Felipe-RA** [reply]
Hola @rosapuertac . Mucho mejor, ya veo de donde salen las 166K filas. Tu dataset es válido, tiene también suficientes valores faltantes y columnas categoricas.

Con respecto a las métricas y el desempeño esperable en producción. Recuerda que el desempeño esperable en produccion se expresa con la métrica de desempeño.

Ahora, tu desempeño en producción va por buen camino, pero necesitas justificarla mejor.  Quién dice que 90%/85% es suficiente? Hay algún manual o criterio ético del que obtienes este número?

 Esta bien que le estes intentando dar una explicación por tu lado, pero este tipo de decisiones deben estar fuertemente respaldadas por **càlculos** o por **expertos en el campo que puedan justificarlo**

Va mejorando tu entrega, sigue así!
